src/components/container/Register/Register.jsx

```diff
@@ -33,86 +22,123 @@ export class Register extends Component {
       username: '',
     },
     validationErrors: [],
-    emptyFields: [],
+    invalidFields: [],
   };
 
   userInputHandler = event => {
-    const { emptyFields } = this.state;
-    const currentEmptyFields = fieldChecker(event, emptyFields);
-    this.setState({
-      userCredentials: {
-        ...this.state.userCredentials,
-        [event.target.name]: event.target.value,
+    const { validationErrors } = this.state;
+    const errors = validate(
+      { [event.target.name]: event.target.value },
+      validationErrors,
+    );
+
+    this.setState(
+      {
+        userCredentials: {
+          ...this.state.userCredentials,
+          [event.target.name]: event.target.value,
+        },
+        validationErrors: errors,
       },
-      emptyFields: currentEmptyFields,
-    });
+      () => {
+        this.handleInvalidFields();
+      },
+    );
   };
 
-  formValidator = event => {
-    event.preventDefault();
-    const { userCredentials } = this.state;
-    const emptyFields = checkAllEmptyFields(userCredentials);
-    if (emptyFields.length) {
-      this.setState({
-        emptyFields,
+  handleInvalidFields = () => {
+    const { validationErrors, userCredentials } = this.state;
+    const errors = [...validationErrors];
+    const fields = Object.keys(userCredentials);
+    const invalidFields = [];
+
+    errors.forEach(error => {
+      fields.forEach(field => {
+        if (error.includes(field) && !invalidFields.includes(field)) {
+          invalidFields.push(field);
+        }
       });
-      return false;
-    }
-    this.setState({ emptyFields: [] });
-    const validationErrors = validateInput(userCredentials);
+    });
+
     this.setState({
-      validationErrors,
+      invalidFields,
     });
-    if (!validationErrors.length) {
-      this.setState({ validationErrors: '' });
-      return true;
-    }
-    return false;
   };
 
-  submitForm = event => {
+  submitHandler = async event => {
     event.preventDefault();
+
+    await this.validateForm();
```

Comment:
is this necessary?

Reply:
When the `submitHandler` is called it is going to submit the inputs immediately instead of waiting for the validation to resolve, so the `await` is necessary there to make sure the error status is returned before proceeding